File: Services/QuestionOptionService.cs
using Dapper;
using Microsoft.Extensions.Options;
using System;
using System.Data;
using System.Linq;
using Microsoft.Data.SqlClient;
using WebApi.Helpers;
using WebApi.Models;
using WebApi.Services.Interfaces;

namespace WebApi.Services
{
    public class QuestionOptionService : IQuestionOptionService
    {
        private readonly ConnectionStrings _connectionStrings;

        public QuestionOptionService(IOptions<ConnectionStrings> connectionStrings)
        {
            _connectionStrings = connectionStrings.Value;
        }

        public dynamic Select(QuestionOptionFilterModel filtro, string tokenUsuario)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_Adm_QuestionOption", new
                    {
                        TypeRequest = "SELECT",
                        filtro.QuestionOptionId,
                        filtro.QuestionId,
                        filtro.ComplementQuestionId,
                        filtro.Rank,
                        filtro.OptionCd,
                        filtro.Description,
                        filtro.OpenMsgBox,
                        filtro.NeedNotes,
                        token_usuario = tokenUsuario
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Create(QuestionOptionCreateModel model, string tokenUsuario)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_Adm_QuestionOption", new
                    {
                        TypeRequest = "INSERT",
                        model.QuestionId,
                        model.ComplementQuestionId,
                        model.Rank,
                        model.OptionCd,
                        model.Description,
                        model.OpenMsgBox,
                        model.NeedNotes,
                        token_usuario = tokenUsuario
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Update(QuestionOptionUpdateModel model, string tokenUsuario)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_Adm_QuestionOption", new
                    {
                        TypeRequest = "UPDATE",
                        model.QuestionOptionId,
                        model.QuestionId,
                        model.ComplementQuestionId,
                        model.Rank,
                        model.OptionCd,
                        model.Description,
                        model.OpenMsgBox,
                        model.NeedNotes,
                        token_usuario = tokenUsuario
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }

        public dynamic Delete(int questionOptionId, string tokenUsuario)
        {
            using (var con = new SqlConnection(_connectionStrings.Default))
            {
                try
                {
                    con.Open();
                    var ret = con.Query("SP_Adm_QuestionOption", new
                    {
                        TypeRequest = "DELETE",
                        QuestionOptionId = questionOptionId,
                        token_usuario = tokenUsuario
                    }, commandType: CommandType.StoredProcedure).ToList();
                    return ret;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                finally
                {
                    con.Close();
                }
            }
        }
    }
}

File: portal-supervisao-360/src/app/core/services/question-option.service.ts
import { Injectable } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { QuestionOptionModel, QuestionOptionCreateModel, QuestionOptionUpdateModel } from '../../shared/models/survey-admin.models';

@Injectable({ providedIn: 'root' })
export class QuestionOptionService {
  constructor(private api: ApiService) {}

  select(filtro: {
    questionOptionId?: number; questionId?: number; complementQuestionId?: number;
    rank?: number; optionCd?: string; description?: string; openMsgBox?: boolean; needNotes?: boolean;
  } = {}): Observable<QuestionOptionModel[]> {
    let params = new HttpParams();
    Object.entries(filtro).forEach(([k, v]) => {
      if (v != null) {
        params = params.set(k, String(v));
      }
    });
    return this.api.getWithParams<QuestionOptionModel[]>('/QuestionOption/Select', params);
  }

  create(model: QuestionOptionCreateModel): Observable<unknown> {
    return this.api.post('/QuestionOption/Create', model);
  }

  update(model: QuestionOptionUpdateModel): Observable<unknown> {
    return this.api.put('/QuestionOption/Update', model);
  }

  delete(id: number): Observable<unknown> {
    return this.api.delete(`/QuestionOption/Delete/${id}`);
  }
}
